mitxgraders/comparers/affine_comparer.py

```diff
@@ -0,0 +1,167 @@
+from collections import namedtuple
+import numpy as np
+from voluptuous import Schema, Required, Any, All, Optional, Range
+from mitxgraders.comparers.baseclasses import CorrelatedComparer
+from mitxgraders.helpers.calc.mathfuncs import is_nearly_zero
+
+def get_affine_fit_error(x, y):
+    """
+    Get total error in a linear regression y = ax + b between samples x and y.
+
+    Arguments:
+        x, y: flat numpy array
+
+    Usage
+    =====
+    Zero error in a linear relationship:
+    >>> x = np.array([2, 5, 8])
+    >>> result = get_affine_fit_error(x, 2*x + 1)
+    >>> round(result, 8)
+    0.0
+    """
+    A = np.vstack([x, np.ones(len(x))]).T
+    _, residuals, _, _ = np.linalg.lstsq(A, y, rcond=-1)
+    return np.sqrt(residuals)
+
+def get_proportional_fit_error(x, y):
```

Comment:
@jolyonb I decided to essentially do 4 separate fits to estimate the error. The `get_affine_error` function fits `y=ax+b`, and the `get_proportional_error` function fits `y=ax`. (The other two never actually do the fits, but still estimate the error.)

This seemed easier than fitting `y=ax+b` once and trying to extract meaning for the 4 separate cases from the coefficients and residuals.

Reply:
Yeah, I think this was a good call.